### PR TITLE
Add a contract customization example to the Required members documentation

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Additional example taken from blog post draft.

### CODE REVIEW

File: docs/standard/serialization/system-text-json/required-properties.md [hunk 1]
```diff
@@ -49,3 +49,30 @@ public class Person
     public int Age { get; set; }
 }
 ```
+
+It is also possible to control whether a property is required via the contract model using the <xref:System.Text.Json.Serialization.Metadata.JsonPropertyInfo.IsRequired?displayProperty=nameWithType> property:
```
Comment (IEvangelist) [inline docs/standard/serialization/system-text-json/required-properties.md:53]:
```suggestion
It's also possible to control whether a property is required via the contract model using the <xref:System.Text.Json.Serialization.Metadata.JsonPropertyInfo.IsRequired?displayProperty=nameWithType> property:
```

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

File: docs/standard/serialization/system-text-json/required-properties.md [hunk 2]
```diff
@@ -49,3 +49,30 @@ public class Person
...
+                {
+                    // strip IsRequired constraint from every property
+                    propertyInfo.IsRequired = false;
+                }
+            }
+        }
+    }
+};
+
+JsonSerializer.Deserialize<Person>("""{"Age": 42}""", options); // serialization now succeeds
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json/required-properties.md:77]:
```suggestion
// Deserialization now succeeds even though the Name property isn't in the JSON payload.
JsonSerializer.Deserialize<Person>("""{"Age": 42}""", options);
```

File: docs/standard/serialization/system-text-json/required-properties.md [hunk 3]
```diff
@@ -49,3 +49,30 @@ public class Person
...
+        Modifiers =
+        {
+            static typeInfo =>
+            {
+                if (typeInfo.Kind != JsonTypeInfoKind.Object)
+                    return;
+
+                foreach (JsonPropertyInfo propertyInfo in typeInfo.Properties)
+                {
+                    // strip IsRequired constraint from every property
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json/required-properties.md:69]:
```suggestion
                    // Strip IsRequired constraint from every property.
```

**Review by gewarren — APPROVED** [review]
Should we update line 11 to say there are 3 ways instead of 2, and then add a third bullet point?

**Pull request merged**